test/integration/apiserver/apiserver_test.go

```diff
@@ -3352,3 +3630,23 @@ func assertManagedFields(t *testing.T, obj *unstructured.Unstructured) {
 func int32Ptr(i int32) *int32 {
 	return &i
 }
+
+func parseFeatureGates(metricsBody string) map[string]int {
+	featureGates := make(map[string]int)
+	scanner := bufio.NewScanner(strings.NewReader(metricsBody))
+	re := regexp.MustCompile(`^kubernetes_feature_enabled{name="([^"]+)",stage="[^"]*"} (\d+)`)
+	for scanner.Scan() {
+		line := scanner.Text()
+		matches := re.FindStringSubmatch(line)
+		if len(matches) == 3 {
```

Comment:
This currently parses each line of /metrics separately so `if len(matches) == 3` occurs only on the correct `kubernetes_feature_enabled...` lines.  If I changed this to error on `if len(matches) != 3` it would error and break the test as it stands ATM (it would always throw an error as there is always a line in the resp.Body from /metrics that isn't a `kubernetes_feature_enabled...` line.

Reply:
I've changed this to use `re.FindAllStringSubmatch` and not loop and now have an error case as suggested

Reply:
+1